src/stubs.ts

```diff
@@ -87,6 +87,33 @@ const isMountedComponent = (
   props: ({ [key: string]: unknown } & VNodeProps) | null | undefined
 ) => isComponent(type) && props && props['ref'] === MOUNT_COMPONENT_REF
 
+const getAsyncComponentName = (
+  type: VNodeTypes,
+  props: ({ [key: string]: unknown } & VNodeProps) | null | undefined
+): string | null => {
+  // Async components will be mounted as AsyncComponentWrapper
+  if (!isMountedComponent(type, props)) return null
+
+  // AsyncComponentWrapper may only contain one component
+  const components = (type as ComponentOptions).components as Record<
+    string,
+    ComponentOptions
+  >
+  if (!components || Object.keys(components).length !== 1) return null
```

Comment:
This seems like it will be a problem. For example

```ts
     it('stubs async component without name', () => {
      const TestComponent = defineComponent({
        components: {
          Foo: {
            template: '<div />'
          },
          MyComponent: AsyncComponentWithoutName,
        },
        template: '<MyComponent/>'
      })

      const wrapper = mount(TestComponent, {
        global: {
          stubs: {
            MyComponent: true
          }
        }
      })

      expect(wrapper.html()).toBe('<my-component-stub></my-component-stub>')
    })
```

Fails, because of

```ts
components: {
          Foo: {
            template: '<div />'
          },
          MyComponent: AsyncComponentWithoutName,
```

We can't rely on the `Object.keys` and getting the first element to capture the name of the component.

What we *can* do, I think, is use the parent component registry. See here: https://github.com/vuejs/vue-test-utils-next/blob/3e85d6ef3da57e5104f2b2ea1e4ca4b2f6ef75fc/src/stubs.ts#L145

I think we can match up using object identity. Something like above. Maybe...

```ts
      if (instance && instance.parent) {
        // try to infer the name based on local resolution
        const registry = (instance.type as any).components
        
        for (const [key, comp] of Object.entries(registry)) {
           if (args[0] === comp) {
             name = key
           }
         }
      }
```

That's untested, but there should be a way to find the component and get the name from the key.